src/prefect/new_flow_engine.py

```diff
@@ -65,8 +68,11 @@ def state(self) -> State:
         return self.flow_run.state  # type: ignore
 
     async def begin_run(self) -> State:
-        state = Running()
-        return await self.set_state(state)
+        new_state = Running()
+        state = await self.set_state(new_state)
+        while state.is_pending():
+            await asyncio.sleep(1)
+            state = await self.set_state(new_state)
```

Comment:
`return state` missing (i think returning will be useful both for unit testing in the future and just because that's the current type hint)